lib/predictions/store.ex

```diff
@@ -5,55 +5,46 @@ defmodule Predictions.Store do
   be retrieved using `fetch/1` for any combination of values specified of
   `fetch_keys`.
   """
+
   use GenServer
 
   require Logger
 
   alias Predictions.Prediction
-  alias Routes.Route
-  alias Schedules.Trip
-  alias Stops.Stop
-  alias Vehicles.Vehicle
-
-  @type fetch_keys :: [
-          prediction_id: Prediction.id_t(),
-          route: Route.id_t(),
-          stop: Stop.id_t(),
-          direction: 0 | 1,
-          trip: Trip.id_t(),
-          vehicle_id: Vehicle.id_t()
-        ]
+  alias Predictions.Store.Behaviour
+
+  @behaviour Behaviour
 
   @spec start_link(Keyword.t()) :: GenServer.on_start()
-  def start_link(opts) do
-    GenServer.start_link(__MODULE__, opts, name: __MODULE__)
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, [], name: __MODULE__)
+  end
+
+  @doc "Deletes predictions associated with the input fetch keys, e.g. clear([route: 'Red', direction: 1])"
+  @impl Behaviour
+  def clear(keys) do
+    GenServer.cast(__MODULE__, {:remove, Enum.map(fetch(keys), & &1.id)})
   end
```

Comment:
suggestion: I would also suggest making `__MODULE__` an argument to this function.

Reply:
Again, we aren't doing DI. We will never need to message a different GenServer.